### PR TITLE
Update comments for the resource weight of `RequestedToCapacityRatio` plugin

### PR DESCRIPTION
**Pull request by chendave**
Signed-off-by: Dave Chen <dave.chen@arm.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by chendave** [comment]
@alculquicondor I guess this a leftover from here: https://github.com/kubernetes/kubernetes/pull/91446/files#r430621496 ?

**Comment by alculquicondor** [comment]
Unfortunately, this would be a breaking change. As the comment suggests, that was fine in alpha, but now we are in beta.
Just remove the TODO, but maybe keep a comment.

**Comment by chendave** [comment]
> Unfortunately, this would be a breaking change. 

Make sense, but this make me a little awkward when I found the API definition here, 
https://github.com/kubernetes/kubernetes/blob/98d3738f6d3e140d193bb3fc5384d93b92fbd562/pkg/scheduler/apis/config/types_pluginargs.go#L159-L162

Should us update the comments there? 

@liggitt What do you think?

**Comment by liggitt** [comment]
> > Unfortunately, this would be a breaking change.
> 
> Make sense, but this make me a little awkward when I found the API definition here,
> https://github.com/kubernetes/kubernetes/blob/98d3738f6d3e140d193bb3fc5384d93b92fbd562/pkg/scheduler/apis/config/types_pluginargs.go#L135-L138
> 
> Should us update the comments there?
> 
> @liggitt What do you think?

what happens if someone specifies a value that exceeds 100 today? if it works as they would expect, we should not break that by tightening validation.

**Comment by chendave** [comment]
> what happens if someone specifies a value that exceeds 100 today?

current algorithm would not break anything, as it will divide the accumulate of weights.

**Comment by chendave** [comment]
> what happens if someone specifies a value that exceeds 100 today? if it works as they would expect, we should not break that by tightening validation.

Got you, I think we'd better to just update those comments.

**Comment by alculquicondor** [comment]
SGTM

**Comment by chendave** [comment]
/retitle  "Update comments for the resource weight of `RequestedToCapacityRatio` plugin"

File: pkg/scheduler/apis/config/types_pluginargs.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ type RequestedToCapacityRatioArgs struct {
 	Shape []UtilizationShapePoint
 	// Resources to be considered when scoring.
 	// The default resource set includes "cpu" and "memory" with an equal weight.
-	// Allowed weights go from 1 to 100.
+	// Allowed weights should be larger than 0.
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/types_pluginargs.go:137]: "Weights should be larger than 0" or "Allowed weights are larger than 0"

File: pkg/scheduler/apis/config/validation/validation_pluginargs.go [hunk 2]
```diff
@@ -240,7 +240,7 @@ func validateFunctionShape(shape []config.UtilizationShapePoint, path *field.Pat
 	return allErrs
 }
 
-// TODO potentially replace with validateResources
+// weight of resource is allowed to exceed 100, this is only applicable to `RequestedToCapacityRatio` plugin for compatibility reason.
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/validation/validation_pluginargs.go:243]: "backwards compatibility"

File: pkg/scheduler/apis/config/types_pluginargs.go (hunk 1 above)
Comment (chendave) [reply pkg/scheduler/apis/config/types_pluginargs.go:137]: Yes, this sounds better.

File: pkg/scheduler/apis/config/validation/validation_pluginargs.go (hunk 2 above)
Comment (chendave) [reply pkg/scheduler/apis/config/validation/validation_pluginargs.go:243]: Done

**Comment by liggitt** [comment]
/approve

doc change for API type lgtm

**Pull request merged**